File: src/coreclr/src/pal/tests/palsuite/filemapping_memmgt/GetProcAddress/test1/testlib.c
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
//

/*=============================================================================
**
** Source: testlib.c (filemapping_memmgt\getprocaddress\test1)
**
** Purpose: Create a simple library containing one function
**          to test GetProcAddress
**
**
**===========================================================================*/
#include "pal.h"

#if WIN32
__declspec(dllexport)
#endif

/**
 * Simple function that returns i+1
 */
int __stdcall SimpleFunction(int i)
{
    return i+1;
}

#if WIN32
int __stdcall _DllMainCRTStartup(void *hinstDLL, int reason, void *lpvReserved)
{
    return 1;
}
#endif

File: src/clr/tests/src/JIT/Methodical/cctor/xassem/testlib.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System;
namespace Precise
{
    public class measure
    {
        public static int a = 0xCC;
    }
    public class test
    {
        public static byte b1 = 0xC;
        public byte b2 = 0xC;
        public static void f()
        {
        }
        public static void f(ref byte b)
        {
            return;
        }

        static test()
        {
            if (measure.a != 0xCC)
            {
                Console.WriteLine("in .cctor(), measure.a is {0}", measure.a);
                Console.WriteLine("FAILED");
                throw new Exception();
            }
            Console.WriteLine("in .cctor(), measure.a is {0}", measure.a);
            measure.a = 8;
            if (measure.a != 8)
            {
                Console.WriteLine("in .cctor() after measure.a=8, measure.a is {0}", measure.a);
                Console.WriteLine("FAILED");
                throw new Exception();
            }
            Console.WriteLine("in .cctor() after measure.a=8, measure.a is {0}", measure.a);
        }
    }

}

